sound/soc/sof/core.c

```diff
@@ -29,13 +29,18 @@ struct snd_sof_pcm *snd_sof_find_spcm_name(struct snd_sof_dev *sdev,
 	struct snd_sof_pcm *spcm = NULL;
 
 	list_for_each_entry(spcm, &sdev->pcm_list, list) {
+		/* match with PCM dai name */
 		if (strcmp(spcm->pcm.dai_name, name) == 0)
 			return spcm;
 
-		if (strcmp(spcm->pcm.caps[0].name, name) == 0)
+		/* match with playback caps name if set */
+		if (strcmp(spcm->pcm.caps[0].name, "") &&
```

Comment:
can't we just use something based on pointers, e.g. if( *str )? strcmp seems overkill here?

Reply:
@plbossart sure I can use strlen instead.